### PR TITLE
Downgrade llvm to 15 to fix jit support

### PR DESCRIPTION
**Pull request by yosifkit**
PostgreSQL doesn't support llvm 16 (https://github.com/docker-library/postgres/issues/1076#issuecomment-1545209600):
- https://www.postgresql.org/message-id/flat/17927-2d7146227560bdf3%40postgresql.org
- https://www.postgresql.org/message-id/flat/17929-71c7a59e1930980b%40postgresql.org

Thank you for your comments on #1076 pointing me in a direction for a fix ❤️. Apologies for any stress and work that this caused 🙇‍♂️. With the added test in https://github.com/docker-library/official-images/pull/14646, we shouldn't run into this issue again.

Fixes https://github.com/docker-library/postgres/issues/1076

### CODE REVIEW

**Comment by yosifkit** [comment]
Hopefully `DOCKER_PG_LLVM_DEPS` should make it easier for PostgreSQL extension installation on Alpine.

If the `LLVM_CONFIG` or `CLANG` variables are necessary/useful to extensions I am willing to move them to `ENV` as well (or maybe scraping them from `pg_config --configure` is enough?).

File: Dockerfile-alpine.template [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{ def llvmver: "15" -}}
```
Comment (tianon) [inline Dockerfile-alpine.template:1]: We should move this down right above `DOCKER_PG_LLVM_DEPS` :eyes:

**Comment by ImreSamu** [comment]
> Hopefully DOCKER_PG_LLVM_DEPS should make it easier for PostgreSQL extension installation on Alpine.

Thank you!
My local [docker-postgis](https://github.com/postgis/docker-postgis/pull/343)  docker build test was successful with the new `DOCKER_PG_LLVM_DEPS`  ! :heavy_check_mark:

File: Dockerfile-alpine.template (hunk 1 above)
Comment (yosifkit) [reply Dockerfile-alpine.template]: Done

**Pull request merged**